add "List of all GitHub PR labels that can influence AutoMerge" (#541)
JuliaRegistries/RegistryCI.jl: docs/src/guidelines.md

very possible for a perfectly good name to not pass the automatic checks and
require manual merging. They simply exist to provide a fast path so that
manual review is not required for every new package.

## List of all GitHub PR labels that can influence AutoMerge

> AutoMerge reads certain labels on GitHub registration pull requests to influence it's decisions.

Reviewer: ```suggestion
AutoMerge reads certain labels on GitHub registration pull requests to influence its decisions.
```

Specifically, these labels are:

* `Override AutoMerge: name similarity is okay`
    * This label can be manually applied by folks with triage-level access to the registry repository.

>     * AutoMerge skips the "name similarity check" on any PRs with this label.

Reviewer: Is it worth mentioning that this is only relevant for "new package" registrations?

Author: ```suggestion
    * AutoMerge skips the "name similarity check" on new package registration PRs with this label.
```

* `Override AutoMerge: package author approved`

>     * This label can be manually applied, but typically is applied by a separate github actions workflow which monitors the PR for comments by the package author, and applies this label if they write "[merge approved]".

Reviewer: ```suggestion
    * This label can be manually applied, but typically is applied by a separate Github Actions workflow which monitors the PR for comments by the package author, and applies this label if they write `[merge approved]`.
```

>     * This label currently only skips the "sequential version number" check in new versions. In the future, the author-approval mechanism may be used for other checks.

Reviewer: ```suggestion
    * This label currently only skips the "sequential version number" check in new versions. In the future, the author-approval mechanism may be used for other checks (on both "new version" registrations and also "new package" registrations).
```

>         * When AutoMerge fails a check that can be skipped by author-approval, it will mention so in the comment, and direct authors to comment "[merge approved]" if they want to skip the check.

Reviewer: ```suggestion
        * When AutoMerge fails a check that can be skipped by author-approval, it will mention so in the comment, and direct authors to comment `[merge approved]` if they want to skip the check.
```
